gloss/Graphics/Gloss/Geometry/Triangulation.hs

```diff
@@ -0,0 +1,388 @@
+
+-- | Utility function for finding first edge down from given event.
+findEdge :: MonotoneEvent -> EdgesWithHelpers -> Maybe Edge
+findEdge p edgesWithHelpers = fst Control.Applicative.<$> Map.lookupLT (Edge (Vertex (eventCootdinates p)) (Vertex (eventCootdinates p))) edgesWithHelpers
+
+-- | Utility function for getting helper.
+getHelper :: Edge -> EdgesWithHelpers -> Maybe MonotoneEvent
+getHelper  =  Map.lookup
+
+-- | Utility function for changing helper of the edge beneath given event.
+changeHelper :: MonotoneEvent -> EdgesWithHelpers -> EdgesWithHelpers
+changeHelper p edgesWithHelpers = let Just e = findEdge p edgesWithHelpers
+                                  in Map.insert e p edgesWithHelpers
+
+-- | Utility function for removing an edge.
+removeEdge :: MonotoneEvent -> MonotoneEvent -> EdgesWithHelpers -> EdgesWithHelpers
+removeEdge p1 p2 = Map.delete (Edge (Vertex (eventCootdinates p1)) (Vertex (eventCootdinates p2)))
+
+-- | Utility function for finging helper of the edge beneath given event.
+findHelper :: MonotoneEvent -> EdgesWithHelpers -> MonotoneEvent
+findHelper p edgesWithHelpers = let Just e = findEdge p edgesWithHelpers
+                                    Just event = getHelper e edgesWithHelpers
+                                in event
+
+-- | Utility function for making a list contatinging tupples of adjacent elements.
+zip3Tail :: [a] -> [(a,a,a)]
+zip3Tail xs = zip3 xs (tail xsForever) (tail $ tail xsForever)
+                where xsForever = xs ++ xs
+
+-- | Utility function for adding an edge to a set of edges.
+addNeighbourSet :: Vertex -> Vertex -> Set.Set (Vertex, Vertex) -> Set.Set (Vertex, Vertex)
+addNeighbourSet v1 v2 s = Set.union s $ Set.fromList [(v1,v2), (v2,v1)]
+
+-- | Marking polyton verteces for x-monotone split.
+markVerteces :: Path -> [MonotoneEvent]
+markVerteces points = map mapFn $ zip3Tail points
+                        where mapFn (p1@(x1, _), p2@(x2, _), p3@(x3, _))
+                                        | (x2 > x1 && x2 > x3) || (x2 == x1 && x2 > x3) || (x2 == x3 && x2 > x1) = if pointsPositiveOrientation (Vertex p1) (Vertex p2) (Vertex p3) == LT then MonotoneEvent p2 END else MonotoneEvent p2 MERGE
+                                        | (x2 < x1 && x2 < x3) || (x2 == x1 && x2 < x3) || (x2 == x3 && x2 < x1) = if pointsPositiveOrientation (Vertex p1) (Vertex p2) (Vertex p3) == LT then MonotoneEvent p2 START else MonotoneEvent p2 SPLIT
+                                        | otherwise =  if x2 > x1  then MonotoneEvent p2 REGULARUP else MonotoneEvent p2 REGULARDOWN
+
+
+-- | Makes a planar graph with fealds bouded by x-monotone polygons and a set of inner edges.
+makeXMonotoneGraph :: Path -> (Vertices, EdgesWithHelpers, Set.Set (Vertex,Vertex))
+makeXMonotoneGraph points = foldl accFn (makeInitialVertexSet points, Map.empty, Set.empty) (sortBy (\(_,a,_) (_,b,_) -> compare (eventCootdinates a) (eventCootdinates b)) $ zip3Tail $ markVerteces points)
```

Comment:
Please reformat this code so it fits in at most 100 columns. Try to follow the coding style in the existing Gloss library.